### PR TITLE
ForwardingService simplifications

### PR DESCRIPTION
**Pull request by msgilligan**
Including:

* ForwardingService: Use awaitRelayed() without extra thenApply()

### CODE REVIEW

**Comment by schildbach** [comment]
Merged 8875a6037438bbafdb22eb1ae34237aba92228a5, 77a1a06db6f600c9a3f8047bb7e13a777120d695, 0c2826661e088a2e8c267a5372f93f8eb2de8287.

**Comment by schildbach** [comment]
5d54e707a6d68db533f728d5596b8e814018e435 looks like it should be 3 separate commits.

**Comment by msgilligan** [comment]
> [5d54e70](https://github.com/bitcoinj/bitcoinj/commit/5d54e707a6d68db533f728d5596b8e814018e435) looks like it should be 3 separate commits.

First one is: PR #2914

**Comment by schildbach** [comment]
Extracted "System.exit rather than throw for wrong number of args" manually.

**Comment by schildbach** [comment]
Merged 5c7290c1778ac190cd7286a5ac82cedaa78358f7.

File: examples/src/main/java/org/bitcoinj/examples/ForwardingService.java [hunk 1]
```diff
@@ -131,10 +118,9 @@ public Address start() {
      */
     @Override
     public void close() {
-        if (kit.isRunning()) {
-            kit.wallet().removeCoinsReceivedEventListener(listener);
```
Comment (schildbach) [inline examples/src/main/java/org/bitcoinj/examples/ForwardingService.java:135]: Where is this listener now removed?

File: examples/src/main/java/org/bitcoinj/examples/ForwardingService.java [hunk 2]
```diff
@@ -65,61 +65,48 @@ public static void main(String[] args) {
...
+
+        // Create and start the WalletKit
+        kit = WalletAppKit.launch(network, new File("."), getPrefix(network), MAX_CONNECTIONS);
+
+        // Add a listener that forwards received coins
+        kit.wallet().addCoinsReceivedEventListener(this::coinForwardingListener);
+
+            // After we start listening, we can tell the user the receiving address
+        System.out.printf("Waiting to receive coins on: %s\n", kit.wallet().currentReceiveAddress());
+            System.out.println("Press Ctrl-C to quit.");
```
Comment (schildbach) [inline examples/src/main/java/org/bitcoinj/examples/ForwardingService.java:109]: Indentation seems off here.

**Review by schildbach — COMMENTED** [review]
Concept ack.

File: examples/src/main/java/org/bitcoinj/examples/ForwardingService.java (hunk 1 above)
Comment (msgilligan) [reply examples/src/main/java/org/bitcoinj/examples/ForwardingService.java:135]: It is not explicitly removed, but it is implicitly removed when everything closes.

File: examples/src/main/java/org/bitcoinj/examples/ForwardingService.java (hunk 2 above)
Comment (msgilligan) [reply examples/src/main/java/org/bitcoinj/examples/ForwardingService.java:109]: Will fix.

File: examples/src/main/java/org/bitcoinj/examples/ForwardingService.java (hunk 1 above)
Comment (msgilligan) [reply examples/src/main/java/org/bitcoinj/examples/ForwardingService.java:135]: I've added back explicit removal.

**Comment by schildbach** [comment]
Merged.

**Pull request closed without merging**